<?xml version="1.0" encoding="utf-8"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda">
    <gazebo>
        <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
            <parameters>$(find chess_manipulator)/config/controllers.yaml</parameters>
        </plugin>
    </gazebo>

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="panda_joint1">
            <command_interface name="position">
                <!-- <param name="min">-2.8973</param> -->
                <!-- <param name="max">2.8973</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint2">
            <command_interface name="position">
                <!-- <param name="min">-1.7628</param> -->
                <!-- <param name="max">1.7628</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint3">
            <command_interface name="position">
                <!-- <param name="min">-2.8973</param> -->
                <!-- <param name="max">2.8973</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint4">
            <command_interface name="position">
                <!-- <param name="min">-3.0718</param> -->
                <!-- <param name="max">-0.0698</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint5">
            <command_interface name="position">
                <!-- <param name="min">-2.8973</param> -->
                <!-- <param name="max">2.8973</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint6">
            <command_interface name="position">
                <!-- <param name="min">-0.0175</param> -->
                <!-- <param name="max">3.7525</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>
        <joint name="panda_joint7">
            <command_interface name="position">
                <!-- <param name="min">-0.0175</param> -->
                <!-- <param name="max">3.7525</param> -->
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />
        </joint>

    </ros2_control>


</robot>